### PR TITLE
Fixed re-wrapping of ByteBuf when sending request body to reactor netty

### PR DESCRIPTION
**Pull request by kushagraThapar**
Direct Port of this PR : https://github.com/Azure/azure-sdk-for-java/pull/9934

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java [hunk 1]
```diff
@@ -137,8 +136,7 @@ private void configureChannelPipelineHandlers() {
                 reactorNettyRequest.header(header.name(), header.value());
             }
             if (restRequest.body() != null) {
-                Flux<ByteBuf> nettyByteBufFlux = restRequest.body().map(Unpooled::wrappedBuffer);
-                return reactorNettyOutbound.send(nettyByteBufFlux);
+                return reactorNettyOutbound.send(restRequest.body());
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:139]: great finding. awesome

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:139]: Thanks Mo :)

**Pull request merged**